### ISSUE TITLE
[BUG] Printer freezes on bigger prints.

### ISSUE THREAD

**Issue by Haxk20** [MarlinFirmware/Marlin#17161]
### Bug Description

We slice the model using Cura and transfer it to SDcard and start the print.
The prints starts out just fine but after some time (It ranges from 10 minutes to several HOURS) it just freezes. No movement on steppers and LCD screen just says the SDcard has been plugged in. But i know that nobody touched the SDcard (I have stood near the printer for the 10 minutes and saw it just stop).
This also weirdly enough happens with USB printing too. LCD says the same thing.
I had this happen on huge print (1.5 day) and it stopped after 16 hours.
I would be kinda OK if it stopped and bed and hotend went to room temp but they still stay heated up. This is a huge fire hazard.

### My Configurations
[Configuration files.zip](https://github.com/MarlinFirmware/Marlin/files/4332947/Configuration.files.zip)
### Steps to Reproduce
1. Start a print
**Expected behavior:** Print starts and finishes just fine resoulting in printed model.

**Actual behavior:** Print starts and after some time freezes.

I also provide the gcode where the print stopped after 17 minutes:
[Gcode.zip](https://github.com/MarlinFirmware/Marlin/files/4332957/Gcode.zip)

**Comment by Haxk20** [comment]
I have just tried deleting the entire SDcard. That yelded one vase mode print for me but failed on second one. @thinkyhead Opened a new issue as you requested.

**Comment by Fantomiaso** [comment]
Have exactly same issue.
SKR1.3 + TMC2208 in uart mode. Stealthchop (LA in firmware is OFF). No display. Anycubic I3 mega hardware.
Prints both from onboard SD and via USB freezes at random point. Heaters, motors and and fan sremain ON. USB-Serial bridge stops responding, but after reconnecting USB I can send commands to Marlin again. I think that problem is in SKR hardware. When I have to test board outside of case any touch of board or wires coud halt it. Inside of metal case board is shielded, but some interference can occur. 
Tomorrow I will try better grounding of board and case. If it will not help, maybe I will add some capacitors on 5V 3V3 lines. In my opinion for reliable operation 12/24V should have 2000-3000uf, and 220uF at least on 5V and 3V3, but thera are only 100uF at 12/24V, 22uF at 3V3 and 33uF at 5V.

My config:
[Config.zip](https://github.com/MarlinFirmware/Marlin/files/4340247/Config.zip)

**Comment by thinkyhead** [comment]
When Marlin is itself paused or stopped it will send messages to host at regular intervals, so that is something to look for.

Marlin will sit and wait if commands are no longer coming in from the SD card or from the host. There's currently no failover mode that deals with a hung up host or an SD card that stops responding. So that would be something new to work on.

**Comment by martend** [comment]
I run skr 1.3 & 1.4 with matching tmc 2208's on 24v and don't run in to these problems. I use BTT stepsticks in uart mode, stealthchop etc. on. I also run one fan over the stepsticks and used the little potmeters to turn the factory current settings of 1.2 down to about 1.0 / 0.8. I observed while doing some prints if the sticks get to hot or there was to low or to much current on an axis (steppermotor) the print would fail. Other functions would continue like nothing happened, including the serial connection. Also a flimsy connection / cable could be the cause, our cheap hardware wants to be connected to good connectors and decent wires :-) I'm half through a 5.5 hour print now, later this evening I'm planning to start a 12h one, with the current Marlin firmware on the board. I'll report how it turned out.

**Comment by Fantomiaso** [comment]
Added 470uF capacitors on 3V3 and 5V lines, 2200uF on 12/24V. Will try some prints for next 24 hours.

**Comment by Fantomiaso** [comment]
Ok, things are getting more interesting. At first look, adding capacitors solved the problem, but after 4-5 hours print stops. Next attempts gives me random durations to failure event. 
Than I  updated firmware to the latest one and while migrating configs, I changed
#define NO_TIMEOUTS 
with value 1000 print goes much further, but I noticed freezes with duration near one second. Finaly print also stopped. It was 2.5 hour print with near 8 freezes. Situation is the same with USB and SD prints. I'll try configs from other printers with working SKR 1.3 and marlin to find what can be wrong. Also I'll get one more SKR, but with 2209 to try.

**Comment by martend** [comment]
@Fantomiaso could you share the gcode file you are printing? Would like to start a print and see what it does on my machine.

**Comment by Fantomiaso** [comment]
One of parts with issues:
[AI3M_Ai3MEGA-S_CaliperHolder_Right.zip](https://github.com/MarlinFirmware/Marlin/files/4350385/AI3M_Ai3MEGA-S_CaliperHolder_Right.zip)
> @Fantomiaso could you share the gcode file you are printing? Would like to start a print and see what it does on my machine.

**Comment by makemerush** [comment]
I previously had this issue on SKR 1.4 Turbo with TMC2130 SPI mode for X/Y/Z and TMC2225 UART for E

Build 2.0.5.1 has fixed it for me! Not sure what changed that would have resolved it though. Recommend the new build if you haven't tried it yet.

**Comment by Haxk20** [comment]
@makemerush The thing is this isnt new issue. Few people had this issue months ago. Then it out of nowhere stopped. And now its back. Its really crazy cause i have just done few prints and all of them went fine. And they werent quick prints. They were 5+ hours prints. So TBH i have no real clue as to where the issue is coming from.
Oh and did i forgot to say that the issue went away without reflashing the firmware or anything ?
Yeah just out of nowhere its gone it seems.
And worst is i never know when it will come back.

**Comment by Fantomiaso** [comment]
Ok, I think I know what is the problem in my case. I started new print from SD via RepetierHost and detouched USB cable from printer (I don't have display at all, so I start prints from host). No freezes, no print stops. I tried different cables and hosts. In all cases, except one, problem turns back. That one case was when I used Odroid XU4 with battery power as host. My version of why that happens is bad USB implementation on hardware level. Maybe power interference, maybe signal. Don't have enough instrumentation to figure out with it. Will try USB print from Odroid next week, when I get stronger battery.

**Comment by Haxk20** [comment]
Here i turned off StealthChop and the print is going fine for 2 hours so far. It is long print so will report if it goes to freeze or finishes.

**Comment by Fantomiaso** [comment]
> 
> 
> Here i turned off StealthChop and the print is going fine for 2 hours so far. It is long print so will report if it goes to freeze or finishes.

There is a known problem with 2208 drivers. An extruder driver in StealthChop mode may freeze when using linear advance. Repraps wiki suggest using:
 #define MINIMUM_STEPPER_PULSE 2
(if notes work try 4)
When 2208 are user in "classic" mode (no uart), only the extruder driver freezes. I suspect that its freeze in the uart mode can lead to deadlocks when the response from this one driver never arrives. If the setting of the pulse duration does not help, it is enough to switch only the extruder driver to the spreadcycle and everything starts to work. But such a problem is not observed with the rest of the drivers and when the linear advanced is turned off.

**Comment by Haxk20** [comment]
Im not running TMCs on Extruder due to this exact reason. I run it only on X and Y

**Comment by Haxk20** [comment]
And the printer just froze.

**Comment by martend** [comment]
Thanks for the file @Fantomiaso, will fire up that one later. 
I printed a nearly 6 hour file yesterday and it went well, but since my cold end was not playing nice I needed to fix that first before starting a long print. I'll start a 12h this evening.
@Haxk20 I noticed you and others use different models of stepperdrivers in your config. Running TMC's only on X and Y. I on the other hand have all axis populated with TMC 2208's and stealthchop on in uart mode. Could it be possible that different stepsticks config don't play nice together at the moment? If I would have any other drivers then TMC's to put in my board I could test along, but I don't have 'm.

**Comment by Haxk20** [comment]
It is very much possible that there is something going on between them. Unfortunately i cannot test with different ones. I given  the original ones to friend.
But TBH it doesnt make sense. You are experiencing the same exact issue as us. Media insert on screen.
OK. Wait i just thought of something. This always started when i changed filament. TBH this at first sounded very crazy as how could filament cause this right ?
Well each filament is different in how much force they can handle before deformation occurs right ?
Well this happens in the extruder also as it need to grab the filament with teeth. And if the filament is grabbed stronger and the deformation is bigger or less then it can push more or less filament.
This in turn can make the motor work harder and require more current. One solution is to up the current but thats not the correct solution as the deformation is different the extruder is pushing different amount of filament. And well PETG and PLA are very different chemically vise. Thus the correct solution is to recalibrate the extruder steps per mm.

(I just came up with that and i may be horribly wrong. So i will tomorrow recalibrate and see what happens. And as safety precaution i will up the current to extruder.)

**Comment by thinkyhead** [comment]
Are you able to force the printer to freeze by turning on heaters and doing fast printing moves? If you switch to "dry run" (or disable the heaters) does the freeze stop happening?

We've recently had an issue reported where an under-powered PSU experienced a voltage drop whenever the machine was printing at high speed with the heaters on (which you kinda need). If you are able to eliminate the problem of a drop in voltage at the input to the board, that will help narrow it down to… something else.

**Comment by Haxk20** [comment]
> Are you able to force the printer to freeze by turning on heaters and doing fast printing moves? If you switch to "dry run" (or disable the heaters) does the freeze stop happening?
> 
> We've recently had an issue reported where an under-powered PSU experienced a voltage drop whenever the machine was printing at high speed with the heaters on (which you kinda need). If you are able to eliminate the problem of a drop in voltage at the input to the board, that will help narrow it down to… something else.

It is well actually absolutely possible its voltage drop. Im technically running the bed at the limit of what PSU can deliver. The issue is i tried to order new PSU but well due to the world in its current state its just not possible to get it in here in the coming month.
But yes i will absolutely try to run it without heating up the printer and dry run it.

**Comment by Haxk20** [comment]
Tested without heating and it finished fine @thinkyhead

**Comment by Haxk20** [comment]
I will heat it up tomorrow and just not add any filament to see if that freezes too.

**Comment by Haxk20** [comment]
@martend @makemerush @Fantomiaso Can you please tell me which slicer do you use ?
Cause i just tried one small print that crashed on me with CURA and now Prusa slice and in prusa slicer it printed. Trying a bigger print that crashed on me 100% of the times i tried it.

**Comment by makemerush** [comment]
Cura 4.5 here

**Comment by Fantomiaso** [comment]
In my case problem is not in software, but in combination of oversimplified schematics of SKR board (with no any protection except fuses) and lack of grounding. I'm living in Belarus and only about ten years ago making grounding in houses became mandatory. Mine is much mor older. I checked voltage between PC housing and 3D-printer and it was floating with peak near 80 volts. Even neon testing screwdriver glows when touching printer or pc, that tells us that it is either phase voltage or strong electromagnetic interference. That voltage latches MCU occasionaly. Same g-codes printed well from SD with USB detouched or via USB odroid single-board pc with battery-powered or fully isolated power source, but failed when printing when USB is connected to my PC.

**Comment by Haxk20** [comment]
@Fantomiaso Can you tell me the slicer tho ?

**Comment by Fantomiaso** [comment]
Slic3r 1.3.1 and Cura 4.5

**Comment by Haxk20** [comment]
Both of them produce freezes ? @Fantomiaso

**Comment by martend** [comment]
@Haxk20 Most of the time Slic3r on a dedicated Linux box.
But since Marlin 2.02? I also started using Cura 4.5.0. Because since that update it seems my printer is on steroids with Slic3r, meaning it's ramping up the speeds like no other. When slicing with Cura it prints with the normal speeds like Slic3r did before.
I know it's a different thing, thought I would mention it anyway since the only thing changed is that Marlin got compiled with the newer version, together with the firmware for my BTT TFT V3.0 
![IMG_20200324_223310](https://user-images.githubusercontent.com/8680187/77479150-873a2e00-6e1f-11ea-93e5-560e526e402a.jpg)
Everything you see in red on the picture was printed the last 2 days, a 12h and 2 x 5h prints. Printed with Cura.

**Comment by Fantomiaso** [comment]
@Haxk20 yeap. 
And as I wrote earlier, it's not they produce freezes (in my case).

**Comment by Haxk20** [comment]
OK so we can exclude slicers i think. What voltage and amperage your PSUs have ?

**Comment by martend** [comment]
220V what goes to a 360w 24V output 20A powerbrick. It's the stock one that comes with Anet ET4.

**Comment by Haxk20** [comment]
I cant see how that could voltage drop. Also thats not 360W thats 480W.
Considering you have good cooling over the PSU i cant see how that could produce a voltage drop. @thinkyhead I cant see how 24V 20A PSU could voltage drop during heavy printing.

**Comment by martend** [comment]
Got the specs from the site, can't look under the machine at the moment it's printing :-)

**Comment by Haxk20** [comment]
Well seems like the OEM cant do simple math. :) Would not be the first or last time sadly.

**Comment by Fantomiaso** [comment]
12V 350W PSU
Consumption: 200W buildplate + 40W hotend when cold. When heated resistanse increases and working current become lower. 60W are for steppers. In worst case I have 50W in reserve.

**Comment by Haxk20** [comment]
Yeah you both have much in reserve. I dont. I run 24V 8.5A (What came with printer). I cant even order a new PSU now due to state the world is in at the moment.

But considering both of you have a lot of power still in reserve it seems unlikely that a voltage drop could appear. Well that is if the PSU is correctly cooled. Now that i think of it. Mine isnt. It could be that the printer heats up the PSU a lot over the hours and that could cause some voltage drop. I will print some feet for the printer and add a cooling fan to the PSU.

**Comment by martend** [comment]
That might do the trick for now. Mail & packages still get delivered to me where I live, so I keep ordering things online :-) Maybe you have access to an old PC PSU those should be beefy enough also.

**Comment by Haxk20** [comment]
Mail still arrives here but Ebay sellers are refusing to send it sometimes. Tried 2 times and both refused to send it due to the COVID-19

**Comment by Fantomiaso** [comment]
> Well that is if the PSU is correctly cooled. Now that i think of it.

By the way very possible. The transformer magnetic core changes its permeability sharply when overheated:
![изображение](https://user-images.githubusercontent.com/62263747/77484248-1a349180-6e3b-11ea-8eb2-52c39c9a2e61.png)

That means that at at temperatures over 90*C magnetic core of transformer requires higher frequency to operate at nominal power (after 120 permeability drops so fast, that transformer stops to work at all). But as switching frequency of PSU is fixed, you can really get powerdrop.

**Comment by Haxk20** [comment]
@Fantomiaso I'm aware. Studying computer science + few classes in electricity :) )
Anyways. Can all of you tell me which SDcard do you use in SKR1.3 ? (Not for printing. But the one where firmware is uploaded)

**Comment by martend** [comment]
I used the original, but now I am on a 32gb Lexar high speed one.

Op wo 25 mrt. 2020 00:48 schreef Haxk20 <notifications@github.com>:

**Comment by Fantomiaso** [comment]
I'm quite sure SKR stores firmware on chip. As for SDs... Any SD I see in that moment) I have plenty of them: from cheap no-name to branded SanDisk for nintendo.

**Comment by amonpaike** [comment]
My experience is this: i have an anet A8 mostly original with a mosfet for heating the bed, and e3d v6 ..
I recently upgraded to marlin 2.0.5.2 after a couple of years, which I didn't upgrade the printer, more or less from version 1.1.6 ...
The printer works at its best, I had it printed heavily for a week 10 -15 hours a day .. it works very well, except that it freeze twice after 4-5  layers (Luckily).
When freeze, the display shows the temperatures at maximum, but the extruder and the bed cool down, and if I try to press the display it does nothing, the only way to make the printer refunction is to remove the power and turn it back on, I also discovered that the power loss restore doesn't even work (another bug).
I relaunch the same identical print and it is processed without other hitches.
Always print from SD card

**Comment by martend** [comment]
@amonpaike pre-heating nozzle and bed might useful here, if you are not already doing that. Also check the temp settings in your slicer. Even since you upgraded, do a PID test for the nozzle and bed. Marlin"s temperature algorithm got stricter and better. 15,16 hours prints are normal for me also, without issues. See picture below.

**Comment by martend** [comment]
21h plus print
![IMG_20200407_150237](https://user-images.githubusercontent.com/8680187/78719994-2cc3c600-7925-11ea-9ef7-0d221fc0485e.jpg)

**Comment by amonpaike** [comment]
> 
> 
> @amonpaike pre-heating nozzle and bed might useful here, if you are not already doing that. Also check the temp settings in your slicer. Even since you upgraded, do a PID test for the nozzle and bed. Marlin"s temperature algorithm got stricter and better. 15,16 hours prints are normal for me also, without issues. See picture below.

Thanks for the advice, I reported my experience because I am almost certain that this is a marlin bug (well who knows what)
I use slic3r PE, and throughout this week of prints I have only had this freeze two times and both times the exact same error, even with similar layer height (not the same amount of filament)
another detail that may be relevant is that throughout the week I used the exact same print settings in slic3r for all files

I can say that I made 13-15 prints in total (from 2-hour prints to 15-hour prints) and twice failed for this freeze

I have not yet updated to the latest firmware version, because as soon as I had this second freeze, today, I relaunched the print and then I came here among the bugs reports to see if something similar had already been reported (or even solved), and here I am to comment.

**Comment by thinkyhead** [comment]
Since "freeze" is a very general thing that computers and software can do for a wide variety of reasons we ask that everyone experiencing a "freeze" post a new issue, filling out the full template, detailing your particular hardware, configuration, and so on, so that your specific "freeze" can be isolated and fixed. If it relates to another issue, that may be determined by further investigation.

**Comment by amonpaike** [comment]
@thinkyhead  at the moment, since I wrote the previous comment, I have updated to the last marlin with a fine tuning with a PID test for the nozzle and bed as suggested by @martend , and I have been updating the parameters of my 3dprinter firmware configuration, for now I have not no longer had any freeze (and I hope it doesn't happen) I hope it is the solution, if time passes and I haven't reported any other freeze again, at least it can be relevant information..

**Comment by martend** [comment]
@amonpaike awesome, glad it solved your issue. Your feedback is welcome and relevant since the other issues in this thread also got solved, this one can be closed if @thinkyhead agrees.

**Comment by Haxk20** [comment]
I have started using octoPrint as when i printed with Cura via USB i saw no issues. And yeah octoPrint is the same. No issues. Also the bonuses are awesome you can get from OctoPrint.

**Comment by Haxk20** [comment]
Forgot to say. Did 64 hours print. No freeze. This was using PETG on 240C and 85C.

**Comment by Haxk20** [comment]
And considering PID tuning solved it for one user so far its almost surely PSU issue then.

**Comment by martend** [comment]
@Haxk20 64 hours.. Nice!! Great Marlin and the board is also working for you and can only agree on Octoprint!

**Comment by amonpaike** [comment]
@martend  I would wait a couple of weeks before closing this bug report, I have an intense week or two of other prints to do these days, and if no freeze will happen and I will not have reported it ... then at least "it is a possible cause and solution "  and a more solid confirmation.

**Comment by xavsbl** [comment]
Got the exact same symptoms on my skr 1.4 turbo and Marlin 2.0.5.3. Didn't try fine PID tuning as I reported previous values of previous tuning but with a bugfix marlin version downloaded around february. So till now and since updating i observed that prints sometimes just stops after 4 to 5 hours printing, but sometimes the printer achieve its duty without problems.
Regarding PSU, it's a 24V/15A here but a somewhat "slim" version of what i'm used to see in CR10 clones (Sunlu S8 to be precise).
I'll report if the pid tuning does the trick.

**Comment by xavsbl** [comment]
It doesn't. Had a 6h print stopping after 2 hours.

> Unexpected error while reading serial port, please consult octoprint.log for details: SerialException: 'device reports readiness to read but returned no data (device disconnected or multiple access on port?)' @ comm.py:_readline:2823
Please see https://faq.octoprint.org/serialerror for possible reasons of this.
Changing monitoring state from "Printing" to "Offline (Error: SerialException: 'device reports readiness to read but returned no data (device disconnected or multiple access on port?)' @ comm.py:_readline:2823)"
Connection closed, closing down monitor

**Comment by xavsbl** [comment]
Here reporting that my issues were not related to this thread. Actually the computer I use to control and monitor my printers has a usb controller issue. 
Sorry.

**Comment by cooldudie2** [comment]
im having the same issue SKR mini E3 Board v1.2 intergrated tmc2209 drivers.printing from sd card Prints will freeze randomly at different points during the print. sometimes it doesn't freeze. can be hard to replicate as its random. Hot end powered and heated bed still powered fully on. the board is unresponsive to any input commands via pronterface or via screen. on;yl solution is to power off and backon again to get it to work.

things i have tried over the last month or so

1. tried cura and prusa slicer to rule out slicing settings/slicer issues
2. Different sd cards , manner of size ranges. always safety ejected from windows before use.
3. tried compiling BTT branch as well as latest marlin 2.0.5.3 same issues.
4. tried the 256k and 512k enviroments made no difference.
5. Linear advance and hybrid theshold modes are disabled to rule out those issues, still freezing.
6. had about 6 successful prints with no problems in a row between 4 -6 hours each problem has returned.
7. tried printing more slowly and dropping aceleration values,
8. tried disabling arc_per_segment, and adaptive step smoothing. stil persists.
9. tried PID tuning the hot end and bed error still persists.   
10. sent board back and returned and bought the board again from another supplier/distributor to rule out a bad board. issue has returned.
11. i havent tried setting it on fire and throwing it in the bin, thats probably going to be my next step.

**Comment by martend** [comment]
How about your cooling? Besides the alu cooling blocks on the steppers, I
also putted on on the lpc1768 chip and I run a fan over the steppers/board.

Op za 30 mei 2020 22:19 schreef cooldudie2 <notifications@github.com>:

**Comment by cooldudie2** [comment]
yes i have the heatsinks on the stepper blocks and a fan running over them for cooling. I have also tried running the tmc debug feature in marlin whilst printing and I do not get any overheat warning triggers for the stepper drivers.

**Comment by cooldudie2** [comment]
problems still persist i have also now ruled out the following
1. the spread jumper is not on the board during testing it is set in firmware, all axis are set to stealthchop mode.

2. I have tried a more powerful ATX power supply unit that is known and good, i have monitored the voltage all the way through the print process and the voltage barley fluctuated 0.2 volts either side of 12v so the power supply issue is ruled out.

3.No TMC prewarn overheat flags were detected/triggered.

i am next going to try running the Extruder in spreadcycle mode only to see if thats the cause as suggested moving the filament like a yo yo in and out constantly may trigger a TMC lockup.

**Comment by amonpaike** [comment]
guys, it hurts me to say that although I had made some refinement adjustments as it was suggested to me a few posts ago, it happened to me again that it's freeze, luckily with the recent firmware that resumes the prints that stop in case of powerloss this problem it is no longer as "risky" as it was before .. but the fact is that it is still here, rarely, but it still happens.

**Comment by taragor** [comment]
I've also had this issue. 
Using SKR mini E3 V2.0 (TMC2209, STM32F1, 24V)
Screen and USB freezes, heaters on.
I've had 5 failed prints by now, and all of them failed during curved perimeters, so I think it might be related to that. 
However that issue occurs only once every ~15-20h (happened always between 1-5h into the print, retrying the same gcode works) for me and I can't trigger it deliberately so trying different configurations is pretty slow.
What didn't fix it for me is:
- disabling Linear Advance
- changing jerk values
- printing from SD/USB
- recompiling/reflashing
The only constants through all my failures were:
- Allways (5/5 failed prints) fails during curved parameters
- Only happens with print speed(cura setting) <=70mm/s (I print most of the time at 70mm/s, however I printed ~20h on 50mm/s and had no fail)

**Comment by boelle** [comment]
@Haxk20 still an issue?

**Comment by boelle** [comment]
Lack of Activity
This issue is being closed due to lack of activity. If you have solved the
issue, please let us know how you solved it. If you haven't, please tell us
what else you've tried in the meantime, and possibly this issue will be
reopened.

**Comment by Haxk20** [comment]
Started using octoprint. Never had this issue. 50 hours print and no
freezes.

**Comment by zsoltkovacs** [comment]
I run into the same problem SKR E3 Mini v1.2 + TMC2209 + Octoprint

If I print via the Octoprint, the board freezes after a while, bed temperature goes to maximum, no activity on steppers, the display free, no response on button, and it drops the serial so OctoPrint detaches and cannot perfect. A perfect fire hazard. I will put back an earlier firmware an see what happens. 

Prints go well if I print from SD card

**Comment by c3D-Dan** [comment]
Another 24 hr print ruined due to inexplicable freezing

**Comment by thinkyhead** [comment]
> Another 24 hr print ruined due to inexplicable freezing

@c3D-Dan — From your report it sounds like a hardware issue. Bad grounding or a board with a bad voltage regulator?

**Comment by amonpaike** [comment]
> 
> 
> Another 24 hr print ruined due to inexplicable freezing

I also risked recently losing a 10-hour print, luckily I activated the resume on power loss, which also works when there are these strange freeze (because you are forced to reset the printer anyway) and I managed not to lose the printed model

**Comment by c3D-Dan** [comment]
Changed to DRV8825, thought the problem was fixed, had a 2 day print complete successfully.  The very next print, failed after 36 hrs.  This is on an STM32F4 platform.

**Comment by c3D-Dan** [comment]
> > Another 24 hr print ruined due to inexplicable freezing
> 
> @c3D-Dan — From your report it sounds like a hardware issue. Bad grounding or a board with a bad voltage regulator?

I highly doubt it.  Printing from new Micro SD Card, this time from the boards own Micro SD Reader. Changed motherboards to a BTT GTR, rewired everything, as above, changed from TMC2209 to DRV8825.  Still freezing for no apparent reason.   Clearly the printer thinks it's finished.  "Motion" menu is available.  Heaters on, LCD still responsive.  Just stopped in its current position.   

Not sure if relevant, but appears to have frozen this time on a Z movement.  Is a vase mode print so can see the freeze occurred at the point at which Z increases.

**Comment by thinkyhead** [comment]
Without more data, I'm afraid there's nothing we can follow up on here.

**Comment by LeBassiste** [comment]
i have an original UM2+ chassis w/ BTT SKR V1.4 TURBO, 4 x TMC5160 drivers running in non-stealthchop.
using the original ULTIMAKER 2+ ULTICONTROLLER, printing from ulticontroller SD-slot, USB on BTT connected to pronterface on host computer for monitoring only. using CURA 4.6.1 w/ GCODE set to MARLIN flavor. 
symptoms:
1. printer sometimes freezes after hours into the print or just after a couple of minutes. always using same GCODE file and same SD-CARD for testing. SD CARD is a SANDISK EXTREME PRO 32GB micro SD and micro-sd adapter, plugged into ulticontroller.
2. when freezing, display still shows "printing..." for about a second, then BTT board does a reset, shows "snow" on display, then bootscreen, then normal start screen. all heaters are then off, parts fan off, and hotend fan starts running, if hotend temp is still over 50°C.
observations:
1. it *seems* that prints are more likely to successfully complete, when i do a power OFF/ON cycle on the printer prior to starting the print.
2. when the printer is switched on and left idling (just doing nothing), no reset will occur (have it right on my desktop while working, so can watch it the entire time).
3. used a thermal camera to check BTT board temperatures. highest is on TMC5160 @ 60 °C, all others (processor, on-board buck regulator and 3.3V LDO) are below 50°C.
4. during all prints, i could never get an OT pre-warn from the drivers.

the configuration I'm using is located here: 
[https://github.com/LeBassiste/Marlin-2.0.5.3-SKR-V1.4-Turbo-UM2Plus/blob/master/Marlin-2.0.5.3-SKR1.4T_UM2plus_V2.zip]

note that i'm using the servo connector on the BTT to control the hotend-fan with PWM.
can do more testing and measurements, but need guidance as to what to look for. can play with config files, but don't have a (useful) programming background.

**Comment by ellensp** [comment]
Please test the [`bugfix-2.0.x` branch](https://github.com/MarlinFirmware/Marlin/archive/bugfix-2.0.x.zip) to see where it stands. If the problem has been resolved then we can close this issue. If the issue isn't resolved yet, then we should investigate further.

**Comment by c3D-Dan** [comment]
> Without more data, I'm afraid there's nothing we can follow up on here.

Let me know what data you require and ill do everything I can to provide it.

**Comment by c3D-Dan** [comment]
> Please test the [`bugfix-2.0.x` branch](https://github.com/MarlinFirmware/Marlin/archive/bugfix-2.0.x.zip) to see where it stands. If the problem has been resolved then we can close this issue. If the issue isn't resolved yet, then we should investigate further.

Am using Bugfix 200005 2.5.3 from memory.    Problem remains.  Printer is very well grounded.

**Comment by ellensp** [comment]
thats ancient...   200006 is current bugfix.

**Comment by LeBassiste** [comment]
testing 200006 right now. did reset eeprom and bed level. 
(have difficulties to test the freeze issue, because as soon as SD-card gets inserted into ulticontroller card slot, info screen shows "media inserted" and starts to irregulary flicker at approx. half-second interval. when trying to select "print from media" menu item w/ click wheel, menu sometimes jumps back to info screen, sometimes stops flickering.
tried #define DOGM_SPI_DELAY_US 20  but to no avail. any other settings i should try on the display config? have to use the ulticontroller though, can't change to other display types.)

ok, managed to get printing something: printer freezes after third layer, display still flickering, menu no longer responding to clickwheel.
edit: board does not go through reset now, as it did with marlin 2.0.5.**3**.

**Comment by LeBassiste** [comment]
could do some more testing with 2.0.5.3 and bugfix 200006.
i actually had two "overlapping" issues, which i did not fully understand before:
**1.** printer would randomly stop mid-print on a print from SD-card (SD-card connected to J14, i.e., EXP1), where the info screen changed from "printing..." to "media inserted" and all heaters and fans kept ON. a capacitor 100nF on EXP2 between J13:7 and J13:9 on the BTT board seems to have fixed that issue. after this, i was able to compare 2.0.5.3 and bugfix 200006 behavior during longer prints, w/o getting this particular behavior anymore.  
**2:** unfortunatley, printer still randomly stops operating on both FW versions after approx. 1 hr. into the print, and goes through a reset.

anything else i could test?

**Comment by thinkyhead** [comment]
A reset is highly unusual and often indicates a hardware problem of some kind. Check your grounding, especially.

**Comment by c3D-Dan** [comment]
https://github.com/MarlinFirmware/Marlin/issues/18358

This *could* and *was*  passed off as "grounding" issues  until @minosg did the ground work to identify the problem on STM32F1 boards.  In his words, he suspects "this issue is not limited to them (STM32F1)".  

Further :
Quoting Minosg
> @c3D-Dan I have suggested some items above. But the core idea is spread around the non critical tasks in the time domain so you don't freeze a handler so long that bits are lost.
> 
> And no this is definitely a core framework bug which occurs because Marlin prepares the right conditions of it.
> 
> Inherently marlin trusts the frameworks to properly implement handlers and they have to trust marlin as the user to be reasonable as to how to use the hardware.
> 
> If I are looking for your platform and the symptoms are the same I would still look if your frameworks uart handler is doing anything for the ORE or overrun bit use case.

Admittedly i'm assuming this is a frame work issue.  Sure my experience is anecdotal at best, however having checked and recheck grounding, replaced at first processors, added heatsinks to MCU and more cooling, replaced TMC drivers with new TMC drivers before replacing the motherboard entirely and replacing TMC drivers with DRV8825s and rewiring everything. Im pretty sure this is an issue beyond the reasonable capacity of an everyday Marlin user.    

@thinkyhead  Is there anything you can do to mitigate the conditions @minosg refers to as a workaround until framework can be fixed properly?   I understand fixing issues with frameworks isn't one of your responsibilities and I super appreciate the insane work you put into this project.  However this issue is clearly wide spread with marlin users who have large build volumes. Not to be ignored, although perhaps overlooked. I think it's safe to say, that if I had a standard 30x30x30 printer, i'd rarely encounter this issue as it would be rare that I could print long enough to encounter the problem.  Perhaps IF this issue was more repeatable on relatively small prints, it would attract more attention. To be clear, the same printer, printing oodles of kinder suprise objects performs perfectly well.  Its not until I (attempt) to print for days on end that this issue rears its ugly head.

**Comment by LeBassiste** [comment]
@ thinkyhead 
apologies for being not quite clear on describing the issue. when the issue happens, the printer simply stops, but still displays "printing..." on the info display. it takes a second or two from there before the printer goes through a reset. (dark screen, boot logo, ...). i actually thought that this behavior is a MARLIN feature  where a watchdog brings the printer back to a safe state after freeze. what am i missing here?
in fact, i concur with you on the probability of a grounding issue. for that matter, i tested the printer with and w/o USB connected to the PC: no change. also, the power supply has a strong (and functioning) PE connection on the mains AC side, as well as the host PC has. on top of it, both printer and PC are operated off the same AC phase.

as a side note: i'm using 4 x TMC5160 with #define TMC_USE_SW_SPI

**Comment by c3D-Dan** [comment]
> as a side note: i'm using 4 x TMC5160 with #define TMC_USE_SW_SPI

I suspected TMC drivers were related to the issue.  Changed with good ol' DRV8825 at 32x.  I did finally manage to get a complete 3 day print, however on the following print, failed at day 2. Heaters on.  Perhaps my first successful print was a fluke?  

I'm completely prepared to test any theory anyone can throw at this without question but within my pay grade, im not sure there's anything else I can do to help.  In order to test and debug, I need the support of those *with* more knowledge and experience to offer more things I can try.

Repeatedly insisting this is a PSU, grounding, heat related, driver related issue is of little help given the time and $ i've spent exploring those ideas to no avail.

**Comment by LeBassiste** [comment]
to whom it may concern, had another freeze this morning, running the bugfix 020006. 2 hrs. into the print, printing from SD card, USB not connected. while displaying "printing...", printer freezes and switches off part fans (nozzle fan keeps running). after approx. 3-4 sec., goes through reset. had successfully printed the same gcode file (5 hrs. total) just yesterday.
any other bugfix approach i should test?
